Crash on startup on M1 Max MacBook Pro

### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

Zed started up OK the first time, but it immediately crashed after I attempted to open a directory with a few Ruby files in it. And now it crashes on startup every time I start it (probably tries to load the same directory.

### Environment

Model: MacBookPro18,2, BootROM 8419.80.7, proc 10:8:2 processors, 64 GB, SMC
Graphics: Apple M1 Max, Apple M1 Max, Built-In
Display: Color LCD, 3456 x 2234 Retina, Main, MirrorOff, Online
Display: DELL U3818DW, 3840 x 1600 (Ultra-wide 4K), MirrorOff, Online
Memory Module: LPDDR5, Hynix
AirPort: spairport_wireless_card_type_wifi (0x14E4, 0x4387), wl0: Nov 30 2022 02:17:16 version 20.10.965.13.8.7.131 FWID 01-1251c18d
Bluetooth: Version (null), 0 services, 0 devices, 0 incoming serial ports
Network Service: Wi-Fi 2, AirPort, en0


### If applicable, add mockups / screenshots to help explain present your vision of the feature

_No response_

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.
If you only need the most recent lines, you can run the `zed: open log` command palette action to see the last 1000.


2023-03-13T18:42:26 [INFO] ========== starting zed ==========
2023-03-13T18:42:26 [INFO] Opening main db
2023-03-13T18:42:26 [INFO] Opening main db
2023-03-13T18:42:26 [INFO] open paths ["/Users/alexk/decisely/osmos"]
2023-03-13T18:42:26 [INFO] Opening main db
2023-03-13T18:42:27 [ERROR] thread 'background-executor-17' panicked at 'failed to set environment variable `""` to `"n3Fv"`: Invalid argument (os error 22)': library/std/src/env.rs:353   0: backtrace::capture::Backtrace::new
   1: Zed::init_panic_hook::{{closure}}
   2: std::panicking::rust_panic_with_hook
   3: std::panicking::begin_panic_handler::{{closure}}
   4: std::sys_common::backtrace::__rust_end_short_backtrace
   5: _rust_begin_unwind
   6: core::panicking::panic_fmt
   7: std::env::_set_var
   8: gpui::executor::any_future::{{closure}}
   9: async_task::raw::RawTask<F,T,S>::run
  10: <futures_lite::future::Or<F1,F2> as core::future::future::Future>::poll
  11: async_io::driver::block_on
  12: std::sys_common::backtrace::__rust_begin_short_backtrace
  13: core::ops::function::FnOnce::call_once{{vtable.shim}}
  14: std::sys::unix::thread::Thread::new::thread_start
  15: __pthread_deallocate

2023-03-13T18:42:27 [ERROR] thread 'main' panicked at 'task has failed': /Users/administrator/.cargo/git/checkouts/async-task-939ec7beeb877e57/341b57d/src/task.rs:368   0: backtrace::capture::Backtrace::new
   1: Zed::init_panic_hook::{{closure}}
   2: std::panicking::rust_panic_with_hook
   3: std::panicking::begin_panic_handler::{{closure}}
   4: std::sys_common::backtrace::__rust_end_short_backtrace
   5: _rust_begin_unwind
   6: core::panicking::panic_fmt
   7: core::panicking::panic_display
   8: core::panicking::panic_str
   9: core::option::expect_failed
  10: futures_lite::future::FutureExt::poll
  11: <gpui::executor::Task<T> as core::future::future::Future>::poll
  12: <futures_util::future::future::shared::Shared<Fut> as core::future::future::Future>::poll
  13: gpui::executor::any_local_future::{{closure}}
  14: async_task::raw::RawTask<F,T,S>::run
  15: <unknown>
  16: <unknown>
  17: <unknown>
  18: <unknown>
  19: <unknown>
  20: <unknown>
  21: <unknown>
  22: <unknown>
  23: <unknown>
  24: <unknown>
  25: <unknown>
  26: <unknown>
  27: <gpui::platform::mac::platform::MacForegroundPlatform as gpui::platform::ForegroundPlatform>::run
  28: gpui::app::App::run
  29: Zed::main
  30: std::sys_common::backtrace::__rust_begin_short_backtrace
  31: std::rt::lang_start::{{closure}}
  32: std::rt::lang_start_internal
  33: _main

--- a/crates/zed/src/main.rs
+++ b/crates/zed/src/main.rs
@@ -437,7 +437,7 @@ async fn load_login_shell_environment() -> Result<()> {
         "SHELL environment variable is not assigned so we can't source login environment variables",
     )?;
     let output = Command::new(&shell)
-        .args(["-lic", &format!("echo {marker} && /usr/bin/env")])
+        .args(["-lic", &format!("echo {marker} && /usr/bin/env -0")])
         .output()
         .await
         .context("failed to spawn login shell to source login environment variables")?;
@@ -449,7 +449,7 @@ async fn load_login_shell_environment() -> Result<()> {
 
     if let Some(env_output_start) = stdout.find(marker) {
         let env_output = &stdout[env_output_start + marker.len()..];
-        for line in env_output.lines() {
+        for line in env_output.split_terminator('\0') {
             if let Some(separator_index) = line.find('=') {
                 let key = &line[..separator_index];
                 let value = &line[separator_index + 1..];

